dagql/server.go

```diff
@@ -886,12 +936,14 @@ func (sel Selector) AppendTo(id *call.ID, spec FieldSpec) *call.ID {
 			// we don't include null arguments, since they would needlessly bust caches
 			continue
 		}
+		var isSensitive bool
 		if arg, found := spec.Args.Lookup(arg.Name); found && arg.Sensitive {
-			continue
+			isSensitive = true
```

Comment:
This logic around sensitive args changed because it was previously just not including the sensitive arg in the ID at all, which no longer works in the new logic where we pass around IDs rather than Selectors in more places.

The fix works by migrating the hiding of sensitive values in the `call/` pkg. It results in the same behavior as before in that an ID will never serialize (via `Display`, `Encode`, etc.) sensitive args and won't include them in digests. The difference is that the sensitive arg can still be carried around on the ID in memory.